beacon-chain/blockchain/process_block.go

```diff
@@ -543,9 +543,13 @@ func (s *Service) handleEpochBoundary(ctx context.Context, postState state.Beaco
 			return err
 		}
 	} else if postState.Slot() >= s.nextEpochBoundarySlot {
+		s.headLock.RLock()
```

Comment:
One option to reduce the lock surface:
```go
s.headLock.RLock()
st := s.head.state
s.headLock.RUnlock()
if err := reportEpochMetrics(ctx, postState, st)...  
```

Reply:
Nice catch! updated in ac7c9b794acd6ca1e38ca11f0dab19e559015283